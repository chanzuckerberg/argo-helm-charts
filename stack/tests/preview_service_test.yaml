suite: test preview service for rollout
templates:
  - preview_service.yaml
tests:
  - it: should not render a preview service when deploymentKind is Deployment
    set:
      global:
        deploymentKind: Deployment
      services:
        service1:
          image:
            repository: "your-app"
            tag: "latest"
    asserts:
      - hasDocuments:
          count: 0

  - it: should work and render a Preview Service with default values
    set:
      global:
        deploymentKind: Rollout
        rollout: {}
      services:
        service1:
          image:
            repository: "your-app"
            tag: "latest"
          args: ["arg1", "arg2"]
          command: ["command1", "command2"]
          autoscaling:
            enabled: true
            minReplicas: 2
            maxReplicas: 10
            maxUnavailable: 1
          replicaCount: 2
          sidecars:
            - name: sidecar1
              image: "alpine:latest"
            - name: sidecar2
              image: "sidecar2:latest"
          initContainers:
            - name: initContainer1
              image: "alpine:latest"
              command: ["echo", "Hello World"]
          service:
            port: 3000
    asserts:
      - matchSnapshot: {}
      - isKind:
          of: Service
      - equal:
          path: metadata.annotations["argocd.argoproj.io/sync-options"]
          value: ServerSideApply=false
  