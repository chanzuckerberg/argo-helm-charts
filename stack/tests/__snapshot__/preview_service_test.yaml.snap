should work and render a Preview Service with default values:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: ServerSideApply=false
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: stack
        app.kubernetes.io/service: release-name-stack-service1
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: stack-2.16.0
      name: release-name-stack-service1-preview
    spec:
      ports:
        - name: http
          port: 3000
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/service: release-name-stack-service1
        rollouts-pod-template-hash: ClusterIP
      type: ClusterIP
